lexers/j/jasmin.go

```diff
@@ -0,0 +1,44 @@
+package j
+
+import (
+	"math"
+	"regexp"
+
+	. "github.com/alecthomas/chroma" // nolint
+	"github.com/alecthomas/chroma/lexers/internal"
+)
+
+var (
+	jasminAnalyserClassRe       = regexp.MustCompile(`(?m)^\s*\.class\s`)
+	jasminAnalyserInstructionRe = regexp.MustCompile(`(?m)^\s*[a-z]+_[a-z]+\b`)
+	jasminAnalyserKeywordsRe    = regexp.MustCompile(
+		`(?m)^\s*\.(attribute|bytecode|debug|deprecated|enclosing|inner|interface|limit|set|signature|stack)\b`)
+)
+
+// Jasmin lexer.
+var Jasmin = internal.Register(MustNewLexer(
+	&Config{
+		Name:      "Jasmin",
+		Aliases:   []string{"jasmin", "jasminxt"},
+		Filenames: []string{"*.j"},
+	},
+	Rules{
+		"root": {},
+	},
+).SetAnalyser(func(text string) float32 {
+	var result float64
+
+	if jasminAnalyserClassRe.MatchString(text) {
+		result += 0.5
+
+		if jasminAnalyserInstructionRe.MatchString(text) {
+			result += 0.3
+		}
+	}
+
+	if jasminAnalyserKeywordsRe.MatchString(text) {
+		result += 0.6
+	}
+
+	return float32(math.Min(result, float64(1.0)))
```

Comment:
Returning a number > 1 might not do any harm. Maybe it is even intended here?

Reply:
We don't know. Let's see what happens with PR I submitted and if we don't get any response I'll change back to the original logic.

Reply:
Ok, let's wait for the pygments PR. But I would suggest to use our general approach for cases of uncertainties and stick to logic of wakatime python client for now.

We cannot go wrong with having the old pygments logic here, because this is what's currently in use in the wakatime python client. And I could imagine, that the pygments PR will hang for some time.

Reply:
Agreed and already reverted the usage of `min()` to the python's logic. Also I'll follow up the PR I submitted to Pygments.